src/lib/xmlparse.rs

```diff
@@ -1599,9 +1527,9 @@ impl XML_ParserStruct {
         parser.m_nsAttsPower = 0;
         parser.m_protocolEncodingName = NULL as *const XML_Char;
 
-        if let XML_ParserStruct {ref m_mem, ref mut m_tempPool, ref mut m_temp2Pool, ..} = &mut *parser {
-            m_tempPool.init(m_mem);
-            m_temp2Pool.init(m_mem);
+        if let XML_ParserStruct {ref mut m_tempPool, ref mut m_temp2Pool, ..} = &mut *parser {
```

Comment:
You should be able to remove this if let binding now. I added it so that the `m_tempPool` field could be used mutably along with `m_mem` being used immutably in the same line

Reply:
Btw, to avoid a compiler warning about the irrefutable if-let pattern, that let binding could have been:

```rust
{
  let XML_ParserStruct {ref mut m_tempPool, ref mut m_temp2Pool, ..} = &mut *parser;
  // do stuff..
}
```

Reply:
That whole conditional is gone now, but the point about the let binding is good to know.